src/script/storage/StorageService.ts

```diff
@@ -358,12 +369,26 @@ export class StorageService {
       throw new z.error.StorageError(z.error.StorageError.TYPE.NO_DATA);
     }
 
-    try {
-      const newKey = await this.engine.updateOrCreate(storeName, primaryKey, entity);
-      return newKey;
-    } catch (error) {
-      this.logger.error(`Failed to put '${primaryKey}' into store '${storeName}'`, error);
-      throw error;
+    if (this.isTemporaryAndNonPersistant) {
```

Comment:
Can we switch on the lower level engine type rather than on the high level feature flag.
The condition to manually trigger the event listeners is not whether the client is temporary but rather if the engine used doesn't support hooks natively.
Let's reflect that idea in the code

Reply:
Good idea, we can determine that using a check like `this.engine instanceof MemoryEngine`.